contracts/connectors/loantoken/LoanTokenLogicStandard.sol

```diff
@@ -285,14 +294,13 @@ contract LoanTokenLogicStandard is LoanTokenSettingsLowerAdmin {
 		uint256 _allowanceAmount
 	) internal returns (bool) {
 		if (_allowanceAmount != uint256(-1)) {
-			require(_value <= _allowanceAmount, "14");
-			allowed[_from][msg.sender] = _allowanceAmount.sub(_value);
+			allowed[_from][msg.sender] = _allowanceAmount.sub(_value, "14");
 		}
 
-		uint256 _balancesFrom = balances[_from];
-		require(_value <= _balancesFrom && _to != address(0), "14");
+		require(_to != address(0), "15");
 
-		uint256 _balancesFromNew = _balancesFrom.sub(_value);
+		uint256 _balancesFrom = balances[_from];
+		uint256 _balancesFromNew = _balancesFrom.sub(_value, "16");
 		balances[_from] = _balancesFromNew;
```

Comment:
`balances[_from] = _balancesFrom.sub(_value, "16");`

Reply:
_balancesFromNew and _balancesToNew are used in L313-L314:
`_updateCheckpoints(_from, _balancesFrom, _balancesFromNew, _currentPrice);`
it is cheaper to read from a memory than from the storage, right?